app/controllers/api_users_controller.rb

```diff
@@ -54,8 +54,13 @@ def load_and_authorize_api_user
 
   def api_user_params
     UserParameterSanitiser.new(
-      user_params: params.require(:api_user),
+      user_params: permitted_user_params(params.require(:api_user)).to_h,
```

Comment:
Unsure about the `to_h` here and in a couple of other places in this changeset. I'd be tempted to handle this type conversion elsewhere. Could potentially get `UserParameterSanitiser` to support the new Params type or introduce an object to go from params to hash and name it something, so it's clearer what is happening.

Reply:
To convert to a hash you need to be sure the parameters have been permitted already - which is why i decided to do it in the controller. You could permit them but still have `UserParameterSanitiser` handle the conversion, but as that conversion removes non-permitted parameters it feels like it could be some hidden behavior not immediately obvious from within the controller